src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
Hmm. An issue with unconditionally showing "Find Definition" is that many regions don't support that. I suppose that this is also going to be an issue for holes, right?

Perhaps the best way forward is to have some regexp-based heuristics for deciding whether the click was in a hole or on an identifier, and then make the appropriate requests. I'll take a look at it today and revise the PR.